rudder-agent/SOURCES/patches/fusioninventory/0020-correct-rudder-inventory-task.patch

```diff
@@ -0,0 +1,169 @@
+Correct the Rudder.pm inventory, to be compatible with Windows and add the server-roles if necessary
+
+--- fusioninventory-agent/lib/FusionInventory/Agent/Inventory.pm	2016-08-30 18:14:54.524244071 +0200
++++ fusioninventory-agent/lib/FusionInventory/Agent/Inventory.pm	2016-08-30 18:15:06.772291754 +0200
+@@ -68,7 +68,7 @@
+     PROCESSES        => [ qw/USER PID CPUUSAGE MEM VIRTUALMEMORY TTY STARTED
+                              CMD/ ],
+     REGISTRY         => [ qw/NAME REGVALUE HIVE/ ],
+-    RUDDER           => [ qw/AGENT UUID HOSTNAME/ ],
++    RUDDER           => [ qw/AGENT UUID HOSTNAME SERVER_ROLE/ ],
+     SLOTS            => [ qw/DESCRIPTION DESIGNATION NAME STATUS/ ],
+     SOFTWARES        => [ qw/COMMENTS FILESIZE FOLDER FROM HELPLINK INSTALLDATE
+                             NAME NO_REMOVE RELEASE_TYPE PUBLISHER
+
+--- fusioninventory-agent/lib/FusionInventory/Agent/Task/Inventory/Generic/Rudder.pm	2016-08-30 18:05:06.181949600 +0200
++++ fusioninventory-agent/lib/FusionInventory/Agent/Task/Inventory/Generic/Rudder.pm	2016-08-30 18:13:54.848011697 +0200
+@@ -7,8 +7,10 @@
+ 
+ use FusionInventory::Agent::Tools;
+ 
++my $uuid_hive = ($OSNAME eq 'MSWin32') ? 'C:\Program Files\Rudder\etc\uuid.hive' : '/opt/rudder/etc/uuid.hive';
++
+ sub isEnabled {
+-    return -r '/opt/rudder/etc/uuid.hive';
++    return -r $uuid_hive;
+ }
+ 
+ sub doInventory {
+@@ -17,23 +19,28 @@
+     my $inventory = $params{inventory};
+     my $logger    = $params{logger};
+ 
+-    # get Rudder UUID
+-    my $Uuid = getFirstLine(
+-        logger => $logger, file => '/opt/rudder/etc/uuid.hive'
++    # Get the Rudder UUID
++    my $uuid = getFirstLine(
++        logger => $logger, file => "$uuid_hive"
```

Comment:
in perl variables don't need another indirection level to be protected. You can remove those ""